LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/JoinRequestMessageHandler.cs

```diff
@@ -238,29 +240,33 @@ async Task ProcessJoinRequestAsync(LoRaRequest request)
                         Logger.Log(devEUI, $"twin RX1 offset DR value is not within acceptable values", LogLevel.Error);
                     }
 
-                    ushort rxDelay = 0;
-                    if (Region.IsValidRXDelay(loRaDevice.DesiredRXDelay))
-                    {
-                        rxDelay = loRaDevice.DesiredRXDelay;
-                    }
-                    else
-                    {
-                        Logger.Log(devEUI, $"twin RX delay value is not within acceptable values", LogLevel.Error);
-                    }
+                // The followind DesiredRxDelay is different than the RxDelay to be passed to Serialize function
+                // This one is a delay between TX and RX for any message to be processed by joining devie
+                // The field accepted by Serialize method is an indication of the delay (compared to receive time of join request)
+                // of when the message Join Accept message should be sent
+                ushort desiredRxDelay = 0;
+                if (loRaDevice.DesiredRXDelay != 0 && Region.IsValidRXDelay(loRaDevice.DesiredRXDelay))
```

Comment:
Nit: we could include this != 0 check in the isvalidRXDelay, 0 is never an acceptable value